legion/engine/llri-vk/utils.cpp

```diff
@@ -161,7 +161,7 @@ namespace llri
             std::vector<VkQueueFamilyProperties> properties(propertyCount);
             vkGetPhysicalDeviceQueueFamilyProperties(physicalDevice, &propertyCount, properties.data());
 
-            for (size_t i = 0; i < propertyCount; i++)
+            for (uint32_t i = 0; i < propertyCount; i++)
```

Comment:
nothing particularly about this line, but i just realised that above this line you use UINT_MAX for uint32_t. I'm not sure UINT_MAX is guaranteed to be 32bit on every architecture and compile target. since it's the max value of `unsigned int` which is not guaranteed to always be 32 bits. I think using numerical limits might be more correct in this case

Reply:
seems like a good moment to fix all UINT_MAX usages, gimme a few mins